index.js

```diff
@@ -1,15 +1,39 @@
+// Best place to find information on XHR features is:
+// https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpReques
+
+var reqfields = [
+  'responseType', 'withCredentials', 'onerror',
+  'timeout', 'ontimeout', 'onabort', 'onprogress'
+]
+
+// Simple ajax function of parameter argument and oncomplete callback
+// Parameters:
+//  - url: string, required
+//  - headers: object
+//  - body: string (sets content type if not set in headers)
+//  - method: 'GET', 'POST', etc. Defaults to 'GET' or 'POST' based on body
+//
+// The following parameters are passed onto the xhr object.
+// The caller is responsible for compatibility checking.
+//  - responseType: string, various compatability, see xhr docs for enum options
+//  - withCredentials: boolean, IE10+, CORS only
+//  - onerror: callback, IE8+ (confirm?), CORS only
+//  - timeout: long, ms timeout, IE8+
+//  - ontimeout: callback, IE8+
+//  - onabort: callback
+//  - onprogress: callback, IE10+
+//
+// Returns the XHR object. So you can call .abort() or other methods
 exports.ajax = function (params, callback) {
-  if (typeof params == 'string') params = {url: params}
```

Comment:
I saved 6 bytes gzipped by moving this to `req.open`. But we might have trouble if people use a library that monkey patches the string object to have `body`, `headers`, `method`, etc.